### DISCUSSION TITLE
Matcher.receive如何正确使用

### DISCUSSION THREAD

**Discussion by DogAddan** [nonebot/discussions#339] [问答] [answered]
接收不到新的事件 然后当前事件响应器就结束了 不知道正确的用法该怎么写

**Comment by yanyongyu** [comment, marked as answer]
https://v2.nonebot.dev/docs/next/tutorial/plugin/create-handler#%E4%BD%BF%E7%94%A8-receive-%E8%A3%85%E9%A5%B0%E5%99%A8

**Reply by DogAddan** [reply]
receive我试了 我想要收到notice的事件 但是收不到

**Reply by yanyongyu** [reply]
默认只接收message类型，如果你需要接收其他任何类型的消息，你需要自定义type_updater

**Reply by DogAddan** [reply]
我原本是消息事件(message)的响应器 然后通过type_updater改成通知事件类型(notice)的了 但是后面也没法接受到通知类型的事件 这么做不是行的吗  还是是我少配了什么

**Reply by yanyongyu** [reply]
日志能否截个图

如果你已经修改为了notice类型，将会接收相同用户的notice事件，你可以确认下通知事件的主体session id是否是相同用户。或者你可以通过permission_updater来自定义权限更新

**Reply by DogAddan** [reply]
找到了 是session id不同 谢谢大佬
